tests/clone3-report-ns-id.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Check appending auxstr of clone3(2) when --namespace=new is given
+ *
+ * Copyright (c) 2024 The strace developers.
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ */
+
+#include "tests.h"
+#include "xmalloc.h"
+
+#include <errno.h>
+#include <limits.h>
+#include <linux/sched.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#include "scno.h"
+
+int
+main(void)
+{
+	skip_if_unavailable("/proc/self/ns/user");
+
+	struct clone_args arg = { .flags = CLONE_NEWUSER };
+	int pid = syscall(__NR_clone3, &arg, sizeof(arg));
+	if (pid < 0) {
+		if (errno == ENOSYS)
+			perror_msg_and_skip("clone3");
+		else
+			perror_msg_and_fail("clone3");
```

Comment:
There could be other errors besides ENOSYS, for example, when creating user namespaces is not permitted, EPERM is returned.  I think we can just skip the test when simple clone3(CLONE_NEWUSER) returns an error.

Reply:
fixed locally